cmd/registry/main.go

```diff
@@ -73,29 +77,7 @@ func init() {
 	SIGNING_KEY = []byte(viper.GetString("SIGNING_KEY"))
 }
 
-func main() {
-	defer func() {
-		_ = log.Sync()
-	}()
-
-	log.Info("Setting up DB Connection")
-	db := connectdb.MakeDBConnection(log, arangodbHost, arangodbCred)
-	log.Info("DB connection established")
-
-	lis, err := net.Listen("tcp", fmt.Sprintf(":%v", port))
-	if err != nil {
-		log.Fatal("Failed to listen",
-			zap.String("address", port), zap.Error(err))
-	}
-
-	auth.SetContext(log, SIGNING_KEY)
-	s := grpc.NewServer(
-		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
-			grpc_zap.UnaryServerInterceptor(log),
-			grpc.UnaryServerInterceptor(auth.JWT_AUTH_INTERCEPTOR),
-		)),
-	)
-
+func SetupSettingsClient(log *zap.Logger, db driver.Database, s *grpc.Server) {
```

Comment:
function doesn't do what it says it is doing (called SetupSettingsClient, but it does register AccountsService Server too)
server registration must be outside this function